src/datatype.rs

```diff
@@ -166,6 +166,15 @@ impl DataType {
     }
 }
 
+impl PartialEq<&str> for DataType {
```

Comment:
Why not `T: AsRef<str>`? It'll be more useful I think

Reply:
Thanks for the hint, I tried it out.
If I got you right, this seems not to be legal, because they could impl AsRef<str> for i.e. f64, then we would have two conflicting implementations for PartialEq.
![image](https://github.com/tafia/calamine/assets/10279020/7be15039-bd2f-4399-9e45-325042bd9b0d)